FEM/pcs_dm.cpp

```diff
@@ -828,7 +828,9 @@ double CRFProcessDeformation::Execute(int loop_process_number)
 #if defined(USE_MPI) || defined(USE_PETSC)
 				}
 #endif
-				if (Error <= Tolerance_global_Newton)//testing only relative residual so far
+				if (Error <= Tolerance_global_Newton && Norm <= m_num->nls_abs_residual_tolerance
```

Comment:
Is the criterion too strict?  Since all three conditions have to be satisfied for the convergence.